js/main.js

```diff
@@ -1,2 +1,23 @@
 /* global data */
 /* exported data */
+var $photoURL = document.querySelector('.photourl');
+var $preview = document.querySelector('.preview');
+
+function updateImg(event) {
+  $preview.setAttribute('src', $photoURL.value);
+}
+
+$photoURL.addEventListener('input', updateImg);
+
+var $form = document.querySelector('form');
+
+function updateEntry(event) {
+  event.preventDefault();
+  var entriesObj = {};
+  data.nextEntryId++;
```

Comment:
This object doesn't include any data from the form elements, it needs a property for each form control element in the form.